exporter/fileexporter/file_exporter.go

```diff
@@ -59,55 +58,115 @@ func (e *fileExporter) consumeTraces(_ context.Context, td ptrace.Traces) error
 	if err != nil {
 		return err
 	}
-	buf = e.compressor(buf)
-	return e.exporter(e, buf)
+	_, err = e.exporter.Write(buf)
+	return err
 }
 
 func (e *fileExporter) consumeMetrics(_ context.Context, md pmetric.Metrics) error {
 	buf, err := e.metricsMarshaler.MarshalMetrics(md)
 	if err != nil {
 		return err
 	}
-	buf = e.compressor(buf)
-	return e.exporter(e, buf)
+	_, err = e.exporter.Write(buf)
+	return err
 }
 
 func (e *fileExporter) consumeLogs(_ context.Context, ld plog.Logs) error {
 	buf, err := e.logsMarshaler.MarshalLogs(ld)
 	if err != nil {
 		return err
 	}
-	buf = e.compressor(buf)
-	return e.exporter(e, buf)
+	_, err = e.exporter.Write(buf)
+	return err
 }
 
-func exportMessageAsLine(e *fileExporter, buf []byte) error {
+type lineWriter struct {
+	mutex sync.Mutex
+	file  io.Writer
+}
+
+func NewLineWriter(cfg *Config, file io.Writer) io.Writer {
+	if cfg.Compression == "zstd" {
+		if fw, err := zstd.NewWriter(file); err == nil {
+			// flushing the compressed writer every second.
+			go func() {
+				for {
+					time.Sleep(1 * time.Second)
+					if fw.Flush() != nil {
+						return
+					}
+				}
+			}()
```

Comment:
I suggest replacing lines 100-104 with `file = fw`, then falling through to line 107 to do the same.

Reply:
Thanks for the review! Using this suggestion now